src/Common/src/TypeSystem/Common/WellKnownType.cs

```diff
@@ -40,5 +40,6 @@ public enum WellKnownType
         RuntimeFieldHandle,
 
         Exception,
+        IndexOutOfRangeException,
```

Comment:
I think that looking things up by name in core library is absolutely fine - there is no point in abstracting it through layers of enums, etc. (Also see my other comment - it may better be a helper.)